src/components/TableComponent/index.jsx

```diff
@@ -16,10 +19,26 @@ class TableComponent extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { id } = this.props;
+    const { id, location } = this.props;
+
+    // Handle the case where the table has changed by paginating the new table
     if (id && id !== prevProps.id) {
       this.props.paginateTable();
     }
+
+    // Handle the case where the router has changed query params in the store. This is used when
```

Comment:
nit: This comment is a little confusing. It says, `"...changed query params in the store"` but we're not actually looking at the query params in the store in the below logic.

Reply:
Good catch. Carry over from when I was mapping router state to props.